src/engine/guidance/collapse_scenario_detection.cpp

```diff
@@ -61,6 +61,17 @@ bool isShortAndUndisturbed(const RouteStep &step)
     return is_short && noIntermediaryIntersections(step);
 }
 
+bool isValidUturnCombination(const RouteStepIterator first, const RouteStepIterator second)
+{
+    auto has_uturn = hasModifier(*first, DirectionModifier::UTurn) ||
+                     hasModifier(*second, DirectionModifier::UTurn);
+    auto const from_name_change_into_uturn =
+        hasTurnType(*first, TurnType::NewName) && hasModifier(*second, DirectionModifier::UTurn);
+    auto const uturn_into_name_change =
+        hasTurnType(*second, TurnType::NewName) && hasModifier(*first, DirectionModifier::UTurn);
+    return !has_uturn || from_name_change_into_uturn || uturn_into_name_change;
```

Comment:
Forbidding uturns in general does end up suppressing a lot of u-turns at intersections that are dual carriage ways in both directions. We need to allow name-changes right after / before to handle these correctly.

So we could think about changing the name here. What we want to be collapsable is the last three statements: Uturns are only part of a collapse if a name-change is involved. Otherwise they cannot be part of a multi-hop instruction.